### PR TITLE
[Relay][TOPI] operator All

### PR DESCRIPTION
**Pull request by yongwww**
Add op `All` in TOPI, RELAY and TF frontend, all is used in numpy and tensorflow, need it to support tf model ssd/fast-rcnn/mask-rcnn.  I will change it to relay.contrib.all, please suggest if you guys have any thought.

@yzhliu @icemelon9 @Laurawly @zhiics @jroesch @srkreddy1238 @tqchen

### CODE REVIEW

File: tests/python/frontend/tensorflow/test_forward.py [hunk 1]
```diff
@@ -1455,6 +1455,17 @@ def check_mean(ishape, **kwargs):
     check_mean((10, 8, 16, 32), axis=(2,3))
     check_mean((10, 8, 16, 32), axis=(1,2), keepdims=True)
 
+#######################################################################
+# All
+# ---
+def test_forward_all():
+    """test All"""
```
Comment (jroesch) [inline tests/python/frontend/tensorflow/test_forward.py]:
```suggestion
    """Test the All operator."""
```

**Comment by jroesch** [comment]
LGTM just waiting on someone with more area expertise.

File: src/relay/op/tensor/reduce.cc [hunk 2]
```diff
@@ -353,6 +353,43 @@ Example::
...
+  return ReduceCompute(attrs, inputs, out_type, target, topi::all);
+}
+
+
+RELAY_REGISTER_REDUCE_OP("all")
+.describe(R"code(Computes the logical AND of array elements over given axes.
+
+Example::
+
+  data = [[[ True,  True,  True],
```
Comment (kevinthesun) [inline src/relay/op/tensor/reduce.cc]: Does this operator allow numerical type? If so. we'd better update the doc.

**Comment by kevinthesun** [comment]
Overall lgtm. A minor comment about doc.

File: python/tvm/relay/op/reduce.py [hunk 3]
```diff
@@ -111,6 +111,37 @@ def sum(data, axis=None, keepdims=False, exclude=False):
...
+        negative it counts from the last to the first axis.
+
+    keepdims : bool
+        If this is set to True, the axes which are reduced are left in the result as
+        dimensions with size one. With this option, the result will broadcast
+        correctly against the input array.
+
+    exclude : bool
+        If `exclude` is true, reduction will be performed on the axes that are
+      NOT in axis instead.
```
Comment (icemelon9) [inline python/tvm/relay/op/reduce.py]: Fix the indent.

File: python/tvm/relay/op/reduce.py [hunk 4]
```diff
@@ -111,6 +111,37 @@ def sum(data, axis=None, keepdims=False, exclude=False):
...
+
```
Comment (icemelon9) [inline python/tvm/relay/op/reduce.py]: Could you also add the example in cpp to python as well?

File: docs/langref/relay_op.rst [hunk 5]
```diff
@@ -309,6 +310,7 @@ Level 6 Definitions
 
 Level 10 Definitions
 --------------------
+.. autofunction:: tvm.relay.all
```
Comment (icemelon9) [inline docs/langref/relay_op.rst]: Let's move `all` to level 4 op since it's a standard numpy op.

File: tests/python/relay/test_op_level10.py [hunk 6]
```diff
@@ -208,10 +208,34 @@ def test_shape_of():
...
+        func = relay.Function([x], z)
+        x_data = np.random.choice([True, False], size=shape)
+        ref_res = np.all(x_data, axis=axis, keepdims=keepdims)
+        for target, ctx in ctx_list():
+            for kind in ["graph", "debug"]:
+                intrp = relay.create_executor(kind, ctx=ctx, target=target)
+                op_res = intrp.evaluate(func)(x_data)
+                tvm.testing.assert_allclose(op_res.asnumpy(), ref_res, rtol=1e-5)
+    verify_contrib_all((2, 3, 4), axis=(0,))
+    verify_contrib_all((2, 3, 4, 5, 6), axis=(2, 3), keepdims=True)
```
Comment (icemelon9) [inline tests/python/relay/test_op_level10.py]: Add axis=None and exclude test case.

Comment (Laurawly) [reply tests/python/relay/test_op_level10.py]: Also axis is negative test cases

**Comment by jroesch** [comment]
@yongwww what's the status on this?

**Comment by yongwww** [comment]
@jroesch I'm on vacation this week, will update the pr to incorporate comments by this weekend

File: src/relay/op/tensor/reduce.cc (hunk 2 above)
Comment (yongwww) [reply src/relay/op/tensor/reduce.cc]: just allow boolean tensor, have updated the doc

**Comment by tqchen** [comment]
@icemelon9 @jroesch please take another look and https://docs.tvm.ai/contribute/code_review.html#approve-and-request-changes-explicitly

**Comment by icemelon9** [comment]
Thanks everyone. This is now merged.

**Pull request merged**